### PR TITLE
test: Stop using BaseEventsTest

### PR DESCRIPTION
**Pull request by lynnagara**
Updates a number of tests to not inherit from BaseEventsTest.
We should move away from needing to have a base class in order to
test a specific dataset, and eventually remove BaseEventsTest.

### CODE REVIEW

File: tests/state/test_state.py [hunk 1]
```diff
@@ -37,13 +35,13 @@ def test_config(self):
...
         @state.memoize(0.1)
-        def rand():
+        def rand() -> float:
             return random.random()
 
         assert rand() == rand()
         rand1 = rand()
         assert rand1 == rand()
-        time.sleep(0.1)
+        time.sleep(1)
```
Comment (evanh) [inline tests/state/test_state.py:44]: Why this change?

**Review by evanh — APPROVED** [review]

File: tests/state/test_state.py (hunk 1 above)
Comment (lynnagara) [reply tests/state/test_state.py:44]: Oops... I thought about changing this since `memoize` is actually supposed to take an int not a float, but I'm going to switch this back since I think it's better to avoid making the tests slower.
